### PR TITLE
Fix #455: Improve Html parser tests

### LINKED ISSUE #455: Improve test cases for HtmlParser and UrlImageParser
HtmlParser and UrlImageParser code was updated to improve the UI, which needs new test cases to check these improvements.

Also, check this comment: https://github.com/oppia/oppia-android/pull/454#pullrequestreview-320146921

### PR DESCRIPTION
**Pull request by manas-yu**
## Explanation
Fix #455
- `testHtmlContentReplace_removesUnwantedNewlines`: Ensures that unwanted newlines are removed from HTML content when parsed.
- `testHtmlContent_withImageTag_trimsLeadingAndTrailingNewlines`: Verifies that leading and trailing newlines are trimmed from HTML content containing an image tag.

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

### CODE REVIEW

**Comment by manas-yu** [comment]
@adhiamboperes, PTAL

File: app/src/sharedTest/java/org/oppia/android/app/parser/HtmlParserTest.kt [hunk 1]
```diff
@@ -427,6 +427,57 @@ class HtmlParserTest {
     assertThat(htmlResult.toString()).endsWith(" ")
   }
 
+  @Test
+  fun testHtmlContentReplace_removesUnwantedNewlines() {
```
Comment (adhiamboperes) [inline app/src/sharedTest/java/org/oppia/android/app/parser/HtmlParserTest.kt]:
```suggestion
  fun testHtmlContentParsing_removesUnwantedNewlines() {
```

File: app/src/sharedTest/java/org/oppia/android/app/parser/HtmlParserTest.kt [hunk 2]
```diff
@@ -427,6 +427,57 @@ class HtmlParserTest {
...
+    }
+
+    assertThat(htmlResult.toString()).isEqualTo(
+      "The counting numbers (1, 2, 3, 4, 5 ….)\nHow to tell whether one counting " +
+        "number is bigger or smaller than another"
+    )
+  }
+
+  @Test
+  fun testHtmlContent_withImageTag_trimsLeadingAndTrailingNewlines() {
```
Comment (adhiamboperes) [inline app/src/sharedTest/java/org/oppia/android/app/parser/HtmlParserTest.kt]:
```suggestion
  fun testHtmlContentParsing_withImageTag_trimsLeadingAndTrailingNewlines() {
```

**Review by adhiamboperes — CHANGES_REQUESTED** [review]
Thanks @manas-yu! These tests look good for the linked PRs. I have left some sugesstions to make the test names more descriptive. Otherwise LGTM.

**Review by adhiamboperes — APPROVED** [review]
Approving and merging.

**Pull request merged**